rust-driver-makefile.toml

```diff
@@ -1,66 +1,393 @@
+//! [dependencies]
+//! wdk-build = { path = "../../../crates/wdk-build", version = "0.1.0" }
+//! ```
+#![allow(unused_doc_comments)]
+
+wdk_build::cargo_make::copy_to_driver_package_folder(
+    wdk_build::cargo_make::get_wdk_build_output_directory().join(format!(
+        "{}.pdb",
+        wdk_build::cargo_make::get_current_package_name()
+    )),
+);
+'''
+
+[tasks.copy-inf-to-package]
+private = true
+dependencies = ["stampinf"]
+script_runner = "@rust"
 script = '''
-copy "%CARGO_MAKE_WORKING_DIRECTORY%\%CARGO_MAKE_CRATE_FS_NAME%.inx" "%OUTPUT_DIR%\package\%CARGO_MAKE_CRATE_FS_NAME%.inf"
-stampinf.exe -f "%OUTPUT_DIR%\package\%CARGO_MAKE_CRATE_FS_NAME%.inf" -d * -a amd64 -c %CARGO_MAKE_CRATE_FS_NAME%.cat -v * -k 1.33 -n
+//! ```cargo
+//! [dependencies]
+//! wdk-build = { path = "../../../crates/wdk-build", version = "0.1.0" }
+//! ```
+#![allow(unused_doc_comments)]
+
+wdk_build::cargo_make::copy_to_driver_package_folder(
+    wdk_build::cargo_make::get_wdk_build_output_directory().join(format!(
+        "{}.inf",
+        wdk_build::cargo_make::get_current_package_name()
+    )),
+);
 '''
 
-[tasks.copypdb]
+[tasks.copy-map-to-package]
+private = true
 dependencies = ["build"]
+script_runner = "@rust"
 script = '''
-cd "%OUTPUT_DIR%"
-copy %CARGO_MAKE_CRATE_FS_NAME%.pdb package\%CARGO_MAKE_CRATE_FS_NAME%.pdb
+//! ```cargo
+//! [dependencies]
+//! wdk-build = { path = "../../../crates/wdk-build", version = "0.1.0" }
+//! ```
+#![allow(unused_doc_comments)]
+
+wdk_build::cargo_make::copy_to_driver_package_folder(
+    wdk_build::cargo_make::get_wdk_build_output_directory().join(format!(
+        "deps/{}.map",
+        wdk_build::cargo_make::get_current_package_name()
+    )),
+);
 '''
 
 [tasks.inf2cat]
-dependencies = ["stampinf"]
+private = true
+dependencies = ["copy-sys-to-package", "copy-inf-to-package"]
+command = "inf2cat"
+args = [
+  "/driver:${WDK_BUILD_OUTPUT_DIRECTORY}/${CARGO_MAKE_CRATE_FS_NAME}_package",
+  "/os:10_NI_X64,10_VB_X64",                                                   # TODO: this should be a parameter
+  "/uselocaltime",
+]
+
+[tasks.generate-certificate-if-needed]
+private = true
+# This script can't be in a `condition_script` block because of https://github.com/sagiegurari/cargo-make/issues/987
+script_runner = "@duckscript"
 script = '''
-inf2cat.exe /driver:"%OUTPUT_DIR%\package" /os:10_NI_X64,10_VB_X64 /uselocaltime /verbose
+out = exec certmgr.exe -put -s WDRTestCertStore -c -n WDRLocalTestCert ${WDK_BUILD_OUTPUT_DIRECTORY}/WDRLocalTestCert.cer
+if not eq ${out.code} 0
+    echo WDRLocalTestCert not found in WDRTestCertStore. Generating new certificate.
+    cm_run_task generate-certificate
+end
 '''
 
-[tasks.infverif]
-dependencies = ["stampinf"]
+[tasks.generate-certificate]
+private = true
+command = "makecert"
+args = [
+  "-r",
+  "-pe",
+  "-a",
+  "SHA256",
+  "-eku",
+  "1.3.6.1.5.5.7.3.3",
+  "-ss",
+  "WDRTestCertStore",                                   # TODO: this should be a parameter
```

Comment:
Future improvement: add support for customizing the name of the Cert Store in `generate-certificate` and `signtool` task